src/pyfreeleh/row/models.py

```diff
@@ -0,0 +1,114 @@
+import dataclasses
+import inspect
+from typing import Any, Dict, Generic, Optional, Type, TypeVar, Union, cast
+
+
+# To differentiate between fields that are not set and Null
+class NotSet:
+    pass
+
+
+T = TypeVar("T")
+
+
+class Field(Generic[T]):
+    _typ: Type[T]
+    _header_name: Optional[str]
+    _field_name: str
+
+    def __init__(self, header_name: Optional[str] = None) -> None:
+        self._header_name = header_name
+
+    def __set_name__(self, _: Any, name: str) -> None:
+        self._field_name = name
+
+        if self._header_name is None:
+            self._header_name = name
+
+    def __get__(self, obj: Any, _: Any) -> Optional[T]:
+        value = getattr(obj._data, self._field_name)
+        return cast(Optional[T], value)
+
+    def __set__(self, obj: Any, value: Optional[T]) -> None:
+        return setattr(obj._data, self._field_name, value)
+
+
+class IntegerField(Field[int]):
+    _typ = int
+
+
+class FloatField(Field[float]):
+    _typ = float
+
+
+class BoolField(Field[bool]):
+    _typ = bool
+
+
+class StringField(Field[str]):
+    _typ = str
+
+
+class PrimaryKeyField(IntegerField):
+    pass
+
+
+class meta(type):
+    def __new__(cls, name: str, bases: Any, dct: Any) -> "meta":
+        new_cls = super().__new__(cls, name, bases, dct)
+
+        # In python3.7 dict ordering is guaranteed based on the insert time.
```

Comment:
How about the ordering of the `bases`?

Reply:
bases depends on the ordering of the class' parent class.